OpenApi parse/merge issues (ionic List templates)

### Expected behavior

Generation of Ionic List templates should merge updated templates without returning errors.

### Actual behavior
When creating ionic List templates from an OpenApi yaml file, the generation of the templates works as expected.
When trying to update the templates (merge) though, the externalProcess returns a 500 error from the server, probably caused by the ts-merger not being able to handle null types.
The OpenApi parser seems to set component types to null, despite them being defined as string or number in the yaml file.

### Steps to reproduce (bug)

#### in Eclipse Plugin

Step 1: Use Full Sample to generate templates from: https://github.com/devonfw/cobigen/wiki/cobigen-openapiplugin#full-example

Step 2: Choose CRUD devon4ng Ionic App/Ionic List from Filter increments and start generation process

Repeat at Step 1

#### in TemplatesGenerationTest devon4j-templates unit test:

Start testAllTemplatesGeneration_OpenApiInput in devon4j-templates project

Following error occurs in several templates:
```
[INFO] Merging    shop-list.page.ts                        FROM ${variables.etoName#lower_case}-list.page.ts       TO ../../devon4ng-ionic-application-template/src/app/pages/shop-list/shop-list.page.ts ...
[ERROR] Sending the request failed. The response message is the following: Internal Server Error
[ERROR] An internal error occurred during generation.
com.devonfw.cobigen.api.exception.MergeException: Unable to merge the patch generated by template X:\Users\...\cobigen-development\master\cobigen-templates\templates-devon4j\src\main\templates\crud_openapi_ionic_client_app\templates\src\app\pages\${variables.etoName#lower_case}-list\${variables.etoName#lower_case}-list.page.ts.ftl into file X:\Users\...\AppData\Local\Temp\cobigen-2052670769312329229\src\app\pages\shop-list\shop-list.page.ts: Execution of the TypeScript merger raised an internal error. Check your file syntax and if error occurs again, please report it on cobigen GitHub.
	at com.devonfw.cobigen.impl.generator.GenerationProcessorImpl.generate(GenerationProcessorImpl.java:389)
	at com.devonfw.cobigen.impl.generator.GenerationProcessorImpl.generate(GenerationProcessorImpl.java:164)
	at com.devonfw.cobigen.impl.generator.CobiGenImpl.generate(CobiGenImpl.java:92)
	at com.devonfw.cobigen.impl.generator.CobiGenImpl.generate(CobiGenImpl.java:74)
	at com.devonfw.cobigen.impl.generator.CobiGenImpl.generate(CobiGenImpl.java:68)
	at com.devonfw.cobigen.maven.GenerateMojo.execute(GenerateMojo.java:181)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```


### Related/Dependent Issues

### Hint
Probably caused by com.reprezen.kaizen.openapi-parser in openapiplugin dependencies

--- a/cobigen/cobigen-tsplugin/pom.xml
+++ b/cobigen/cobigen-tsplugin/pom.xml
@@ -15,7 +15,7 @@
   <properties>
     <plugin.name>${project.artifactId}</plugin.name>
     <server.name>nestserver</server.name>
-    <server.version>1.1.2</server.version>
+    <server.version>1.1.5</server.version>
     <server.url>https\://registry.npmjs.org/@devonfw/cobigen-nestserver/-/cobigen-nestserver-${server.version}.tgz</server.url>
     <server.url.linux>https\://registry.npmjs.org/@devonfw/cobigen-nestserver-linux/-/cobigen-nestserver-linux-${server.version}.tgz</server.url.linux>
     <server.url.macos>https\://registry.npmjs.org/@devonfw/cobigen-nestserver-macos/-/cobigen-nestserver-macos-${server.version}.tgz</server.url.macos>

--- a/cobigen/cobigen-tsplugin/src/test/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMergerTest.java
+++ b/cobigen/cobigen-tsplugin/src/test/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMergerTest.java
@@ -186,4 +186,17 @@ public class TypeScriptMergerTest {
         return returnString;
     }
 
+    /**
+     * Tests if TypeScript can handle null- and undefined-aware types
+     *
+     * @test fails
+     */
+    @Test
+    public void testNullAndUndefinedTypes() {
+        TypeScriptMerger tsMerger = new TypeScriptMerger("tsmerge", false);
+        File baseFile = new File(testFileRootPath + "nullBase.ts");
+        String mergedContents = tsMerger.merge(baseFile, readTSFile("nullPatch.ts"), "UTF-8");
+        assertEquals(false, mergedContents.contains("Not able to merge") || mergedContents.isEmpty());
+    }
+
 }

--- a/cobigen/cobigen-tsplugin/src/test/resources/testdata/unittest/files/nullBase.ts
+++ b/cobigen/cobigen-tsplugin/src/test/resources/testdata/unittest/files/nullBase.ts
@@ -1,0 +1,4 @@
+export class SomeDataList {
+  somedataSearchCriteria: SomeDataSearchCriteria = {
+  };
+}

--- a/cobigen/cobigen-tsplugin/src/test/resources/testdata/unittest/files/nullPatch.ts
+++ b/cobigen/cobigen-tsplugin/src/test/resources/testdata/unittest/files/nullPatch.ts
@@ -1,0 +1,6 @@
+export class SomeDataList {
+  somedataSearchCriteria: SomeDataSearchCriteria = {
+    AnyString: null,
+    AnotherString: undefined,
+  };
+}

